virtcontainers/sandbox.go

```diff
@@ -639,11 +640,15 @@ func (s *Sandbox) createCgroupManager() error {
 		cgroupPath = spec.Linux.CgroupsPath
 
 		// Kata relies on the cgroup parent created and configured by the container
-		// engine, but sometimes the sandbox cgroup is not configured and the container
-		// may have access to all the resources, hence the runtime must constrain the
-		// sandbox and update the list of devices with the devices hotplugged in the
-		// hypervisor.
-		resources = *spec.Linux.Resources
+		// engine by default. The exception is for devices whitelist as well as sandbox-level
+		// CPUSet.
+		resources.CPU = &specs.LinuxCPU{
+			Cpus: spec.Linux.Resources.CPU.Cpus,
```

Comment:
device cgroup must be honoured

Reply:
AFAIU this is still supporting devices (it is just modifying the CPULinux field)?

Not ideal, but it seems that devices are handled 'sideband' in the cgroup-mgr pkg. Take a second look, or can you clarify?

Reply:
Also, do we have any CI tests to verify this? I assume I should at the very least see that the VMM required devices are present in the whitelist?

Reply:
n/m -- I see the issue here - will update  to keep / preserve the resources.Devices structure

Reply:
updated, PTAL @devimc